# SML017: ErrorSwallowing

Any kind of exception/error swallowing or hiding of error details

|Item|Value|
|-|-|
|Category|CodeSmellAnnotations|
|Enabled|True|
|Severity|Warning|
|CodeFix|False|
---

* Improper exception handling
```csharp
catch(SomeExceptionType ex)
{
	_logger.LogError(ex, "some text");
	
	throw; // <-- correct
	throw new SomeCustomException($"Failed for id {id}", ex); // <-- correct
	throw ex; // <-- incorrect; you will lose call stack details
	throw new Exception(ex); // <-- incorrect; wrapping the exception is just meaningless
}
```
---
* If we are not going to process the exception, we should not catch it at all:
 ```csharp
catch(SomeExceptionType) // <-- this whole block is meaningless
{
	throw; 
}
```
---
* Exception swallowing:
```csharp
// this is bad code (except in some very rare cases)
catch (SomeException ex)
{
 // nothing
}
```
---
* Discarding of exception details
```
try
{
	// boom
}
// this is very bad code
catch
{
    return new Result { IsSuccessful = false, Message = "An error occured!" };
}
```


Source:
```cs
[CodeSmell(Kind.ErrorSwallowing, Reason = "Some detais of the error are lost.")]
```